src/ansys/speos/core/sensor.py

```diff
@@ -3742,3 +3742,617 @@ def set_geometries(self, geometries: [List[GeoRef]]) -> Sensor3DIrradiance:
             gr.to_native_link() for gr in geometries
         ]
         return self
+
+
+class SensorXMPIntensity(BaseSensor):
+    """Class for XMP intensity sensor.
+
+    Parameters
+    ----------
+    project
+    name
+    description
+    metadata
+    sensor_instance
+    default_values
+    """
+
+    def __init__(
+        self,
+        project: project.Project,
+        name: str,
+        description: str = "",
+        metadata: Optional[Mapping[str, str]] = None,
+        sensor_instance: Optional[ProtoScene.SensorInstance] = None,
+        default_values: bool = True,
+    ) -> None:
+        if metadata is None:
+            metadata = {}
+
+        super().__init__(
+            project=project,
+            name=name,
+            description=description,
+            metadata=metadata,
+            sensor_instance=sensor_instance,
+        )
+
+        # Attribute gathering more complex intensity type
+        self._type = None
+        self._layer_type = None
+
+        if default_values:
+            # Default values template
+            self.set_type_photometric()
+            self.set_orientation_x_as_meridian()
+            self.set_viewing_direction_from_source()
+            # Default values properties
+            self.set_axis_system().set_layer_type_none()
+
+    @property
+    def nearfield(self) -> bool:
+        """Property containing if the sensor is positioned in nearfield or infinity."""
+        if self._sensor_template.intensity_sensor_template.HasField("near_field"):
+            return True
+        else:
+            return False
+
+    @nearfield.setter
+    def nearfield(self, value):
```

Comment:
if using property/setter methods, you will need to call the responding parameters setter methods as well. 
i.e. for near field as true, then you need set call distance and cell diameter as well. 
otherwise, the default setting will not happen.